### DISCUSSION TITLE
Migration Filename Issue

### DISCUSSION THREAD

**Discussion by Snuupy** [Zibbp/ganymede#194] [Q&A]
Hi again, I thought discussions would be a more appropriate place than opening an issue :)

I'm having some issues migrating my library to the new storage template format (probably because I ran the task when there was a [permissions issue](https://github.com/Zibbp/ganymede/issues/193)). The task failed and ran into some errors, but now I'm unable to migrate older videos to the new format and they are now missing from the catalog unless I check the logs to see what folder name it expects and rename the folder to the one specified in the error for each one manually.

Maybe the filename entry in the database should not update if the task fails.

![image](https://user-images.githubusercontent.com/964538/236444143-757c1c12-50c8-4e88-9fa0-b3aecfe1bdd2.png)

If I check the container logs, I see the following:

`ganymede-nginx     | 2023/05/05 11:20:43 [error] 31#31: *664 open() "/mnt/vods/healthygamer_gg/2022-11-14-1652670221-archive-6f77afe7-67ed-11ed-82d8-0242ac120004/1652670221-web_thumbnail.jpg" failed (2: No such file or directory), client: 192.168.1.235, server: , request: "GET /vods/healthygamer_gg/2022-11-14-1652670221-archive-6f77afe7-67ed-11ed-82d8-0242ac120004/1652670221-web_thumbnail.jpg HTTP/1.1", host: "server.lan:4802", referrer: "http://server.lan:4801/"`

If I rename the folder from `1706254536_c5747fc8-93a6-11ed-a729-0242ac120003/` to `2023-01-13-1706254536-archive-c5747fc8-93a6-11ed-a729-0242ac120003/` manually, I am able to view the video once again.

Problem is, the original folder name does not include the date and I have 60+ archived vods from this specific channel alone. Is there any way I can get the automated migration task to work on my library?

Thanks.

**Comment by Zibbp** [comment]
If you edit a vod that failed to rename in Admin > Vods, are the various paths pointing to the old or new file name?

**Reply by Snuupy** [reply]
all towards new

**Reply by Zibbp** [reply]
Hmm, you could try changing the storage template to match the old file names, run it, change it back to your liking, then run the task again. This should update the database to use the old format for old folder, while updating the new ones to match the same format. Once everything is happy, there should be no problem going back to your previous template. The temporary folder template would be `{{id}}_{{uuid}}`. I haven't tried doing this though, so backup your database first.
